### ISSUE TITLE
Encoding and decimal separator

### ISSUE THREAD

**Issue by houdelou** [eramella/QFXparser#7]
I don't know if you could consider those issues but I ran into two problems while using the library.

**Encoding**
By default, qfx files are read using UTF-8 encoding. Mine was ISO-8859-1 encoded. I had to convert my file to utf-8 to be able to read accented characters correctly. I don't know what would be the good way to do it ? One option would be to add an encoding argument. The other would be to take a StreamReader as argument instead of Stream, which is already encoded. This way, we could use any encoding.

**Separator**
The library is alway taking the default culture settings to parse amounts. My file had dots as decimal separator, so every amount containing dots were not read. I had to set the culture settings for the thread to workaround this problem. It would be nice to specify cultureinfo or the separator as parameter.

Thanks for the library, it was very useful for me.

**Comment by eramella** [comment]
Thank you so much for pointing these issues out. I created this library I guess too much from the US point of view. I would be happy to consider your suggestions.
Couple of ways about it: if you could attach here a sample qfx file from your location so that I can work your suggestion with a concrete example. Or if you feel like sending a PR I will be happy to accept it (still include a sample file so that I can accurately test).
Thank you again .

**Comment by houdelou** [comment]
Thank you for your reply. I'll be happy to help and send a PR and a file !

**Comment by houdelou** [comment]
I looked a little bit more at the files and code. I noticed our files are in the same encoding (USASCII CP1252) and format (decimal separator). If I look at the header, our charset and encoding are also the same. I think why your statements are read correctly is because you don't have any special characters in them. Cause the streamreader used in the code is UTF-8 by default to read a CP1252 file. 

For the separator, the problem is that our system does not have the same cultureinfo. Mine is fr-CA with comma as decimal separator. Yours is probably en-US with period as decimal separator. I suppose that using CultureInfo.InvariantCulture would solve the problem as it would be independent of the system settings.

Of course, we could parse the header to see in which encoding is the file, but since qfx does not seems to have any whitepaper attached to it, it is difficult to parse if we don't have all the possible values of the elements... I'm sure you've run into some problems too because of that ;).

I attach a file as asked.
[test_iso8859-1.qfx.txt](https://github.com/eramella/QFXparser/files/4526040/test_iso8859-1.qfx.txt)
